cpp/cmake/thirdparty/get_nanoarrow.cmake

```diff
@@ -26,6 +45,7 @@ function(find_and_configure_nanoarrow)
     # TODO: Commit hashes are not supported with shallow clones. Can switch this if and when we pin
     # to an actual tag.
     GIT_SHALLOW FALSE
+    PATCH_COMMAND ${CMAKE_COMMAND} -P ${patch_script}
```

Comment:
Using `PATCH_COMMAND` means that we will never search for a local version of `nanoarrow` even when doing a shared build. Is that desired?

Reply:
For now I think that's desirable because without these patches and prior to 0.5.0 nanoarrow's local builds don't even have usable CMake. Once we upgrade to 0.5.0 we'll drop this patch. If we need to reintroduce patching later, we can revisit this question.

Reply:
Works for me